### PR TITLE
Fix bug 2521

### LINKED ISSUE #2521: [BUG] Failed to override Alluxio's properties in spec.master or spec.worker
**What is your environment(Kubernetes version, Fluid version, etc.)**
6e6294b

**Describe the bug**
In cases that I want to set properties of Alluxio, only properties in `AlluxioRuntime.spec.properties` take effect. The properties set in `AlluxioRuntime.spec.master.properties` or `AlluxioRuntime.spec.worker.properties` are ignored.

For example:
```
apiVersion: data.fluid.io/v1alpha1
kind: Dataset
metadata:
  name: hbase
spec:
  mounts:
    - mountPoint: https://mirrors.bit.edu.cn/apache/hbase/stable/
      name: hbase
---
apiVersion: data.fluid.io/v1alpha1
kind: AlluxioRuntime
metadata:
  name: hbase
spec:
  replicas: 1
  properties:
    alluxio.worker.block.heartbeat.interval: "30sec"
    alluxio.master.rpc.executor.keepalive: "30sec"
  master:
    properties:
      alluxio.master.rpc.executor.keepalive: "45sec"
      alluxio.master.rpc.executor.max.pool.size: "300"
  worker:
    properties:
      alluxio.worker.block.heartbeat.interval: "60sec"
      alluxio.worker.block.heartbeat.timeout: "1800sec"
  tieredstore:
    levels:
      - mediumtype: MEM
        path: /dev/shm
        quota: 2Gi
        high: "0.95"
        low: "0.7"
```

The expected properties should be:
- alluxio.master.rpc.executor.keepalive: "45sec"
- alluxio.master.rpc.executor.max.pool.size: "300"
- alluxio.worker.block.heartbeat.interval: "60sec"
- alluxio.worker.block.heartbeat.timeout: "1800sec"

However, what I got in Alluxio Master Pod by `alluxio getConf` is like:
```
$ alluxio getConf | grep "alluxio.master.rpc.executor"
alluxio.master.rpc.executor.keepalive=30sec
alluxio.master.rpc.executor.max.pool.size=1024
```

```
$ alluxio getConf | grep "alluxio.worker.block"
alluxio.worker.block.heartbeat.interval=30sec
alluxio.worker.block.heartbeat.timeout=1hour
```

**What you expect to happen:**
`AlluxioRuntime.spec.master.properties` and `AlluxioRuntime.spec.worker.properties` should override the properties in AlluxioRuntime.spec.properties.

**How to reproduce it**
Follow the steps mentioned above

**Additional Information**

### PR DESCRIPTION
**Pull request by uniqueni**
Signed-off-by: niwang <niwangzz@163.com>

### Ⅱ. Does this pull request fix one issue?

fixes #2521 

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Comment by uniqueni** [comment]
I has been verified locally
image

File: pkg/ddc/alluxio/transform.go [hunk 1]
```diff
@@ -299,6 +299,7 @@ func (e *AlluxioEngine) transformMasters(runtime *datav1alpha1.AlluxioRuntime,
 
 	if len(runtime.Spec.Master.Properties) > 0 {
 		value.Master.Properties = runtime.Spec.Master.Properties
+		e.overrideProperties(runtime, value.Master.Properties)
```
Comment (TrafalgarZZZ) [inline pkg/ddc/alluxio/transform.go:302]: I suggest to reuse utility function [`UnionMapsWithOverride`](https://github.com/fluid-cloudnative/fluid/blob/master/pkg/utils/map.go#L21) to fix this.

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: pkg/ddc/alluxio/transform.go (hunk 1 above)
Comment (uniqueni) [reply pkg/ddc/alluxio/transform.go]: done

**Pull request merged**